web-ui/src/app/components/latest-blocks/latest-blocks.component.ts

```diff
@@ -72,28 +71,16 @@ export class LatestBlocksComponent implements OnInit, OnDestroy {
   }
 
   private onBlockRetrieved(response: Block[]) {
-    this.countNewestBlocks(response);
-    this.latestBlock = response[0];
+    this.newBlocks = this.getNewBlocks(response);
     this.blocks = response;
   }
 
-  private countNewestBlocks(newBlocks: Block[]) {
+  private getNewBlocks(newBlocks: Block[]): Block[] {
     if (!this.blocks) {
```

Comment:
again, this is harder to test than just passing the required argument to the function, please stop using global state.